src/bluetooth/BreachManager.js

```diff
@@ -0,0 +1,189 @@
+import moment from 'moment';
+
+// Default breach creation method.
+const createBreachRecord = (
```

Comment:
Hmm there is the file `createRecord` ⚠️ 

Think we should probably use it (Can edit the func how you need)

Reply:
🙀  wow. Ok thanks I'll give this a go

Reply:
On that note, `createRecord` is getting unwieldy (i.e. 1000+ lines). I'm wondering if it'd be better/plausible to use constructors in the realm classes and what that'd look like, how they'd be used?

[The realm docs](https://realm.io/docs/kotlin/latest/#limitations-models) would suggest it's possible but with minor(?) limitation:
>7. If a custom constructor is provided, a public no-arg constructor must also be present.

Reply:
After looking at `createRecord` I'm not convinced this particular method is an ideal candidate for the file - as this is just building up an object, whereas the general pattern for `createRecord` seems to be to create + insert/update to DB in one go (which is also less clean?).

I briefly tried experimenting with adding a constructor to the TemperatureBreach Realm Object (cos it sounded like a really neat idea) but trying to do a DB update using the constructed object didn't seem to work, I think there's more magic going on behind the scenes I don't understand. Also the docs above seemed to be kotlin related, tried to multi-constructor in JS and got told off haha. Probably doing something wrong but maybe something still worth exploring?  Thought I'd stop digging before spending too long.

Reply:
Yeah I've looked into realm constructors in the past as it would be pretty wicked for di and customizations - but I don't think it's worthwhile looking at them as we seem to be pretty set on sqlite if there were to be any big db moves. You're right about `createRecord`, what you've done is good